### PR TITLE
add rd_kafka to the stack

### PR DESCRIPTION
**Pull request by AntonErmolenko**
assist https://github.com/test-IO/app/pull/5428/files#diff-22d63200ac4f87b849d6927890f0d75aR9, include rd_kafka in the gem.

### CODE REVIEW

File: lib/message_express/bus/rd_kafka.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+      def initialize(kafka_client, topic: nil)
+        @kafka = kafka_client
+        @topic = topic || 'messages'
+      end
+
+      def publish(message)
+        producer_params = {
+          topic: @topic,
+          payload: message,
```
Comment (garciaf) [inline lib/message_express/bus/rd_kafka.rb]: Do not forget our messages have the following structure (I still don't fully understand why though)
```ruby
message = {
  payload: payload,
  name: 'name of the event'
}
```
so in your case 
```
  payload: message.payload
```

File: lib/message_express/bus/rd_kafka.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+          key: "#{@topic}-#{rand(10**10)}"
```
Comment (garciaf) [inline lib/message_express/bus/rd_kafka.rb:18]: Why do you need to define the key? The gem is not taking care of it?

File: lib/message_express/bus/rd_kafka.rb [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+      end
+
+      def subscribe(consumer_group_id:, topic: @topic)
+        consumer = @kafka.consumer(group_id: consumer_group_id)
+        consumer.subscribe(topic)
+
+        trap('TERM') { consumer.stop }
+
+        consumer.each do |kafka_message|
+          Rails.logger.info "Message: #{kafka_message}"
```
Comment (garciaf) [inline lib/message_express/bus/rd_kafka.rb:32]: You don't need to mark this message as processed?

Comment (garciaf) [reply lib/message_express/bus/rd_kafka.rb:32]: There is a way to know if a message have been listened already?

I also think we should do something with the message. 
```ruby
message_payload = JSON.parse(kafka_message.payload)
message = Message.new(kafka_message.key, message_payload)
yield(message)
```

**Review by garciaf — COMMENTED** [review]
I try to make sure both implementation have a similar logic

File: lib/message_express/bus/rd_kafka.rb (hunk 2 above)
Comment (AntonErmolenko) [reply lib/message_express/bus/rd_kafka.rb:18]: No it's not in this case. At first I'd thought that also... but errors arose, and I had to add it.

File: lib/message_express/bus/rd_kafka.rb (hunk 3 above)
Comment (AntonErmolenko) [reply lib/message_express/bus/rd_kafka.rb:32]: this subscriber is for the test proposes only, other applications will process them.

File: lib/message_express/bus/rd_kafka.rb (hunk 1 above)
Comment (AntonErmolenko) [reply lib/message_express/bus/rd_kafka.rb]: ok thanks!

**Comment by AntonErmolenko** [comment]
I think this pr should be closed, because of different purposes of Kafka's. CC @garciaf

**Pull request closed without merging**